service/debugger/debugger.go

```diff
@@ -1414,6 +1414,19 @@ func (d *Debugger) EvalVariableInScope(goid, frame, deferredCall int, symbol str
 	return s.EvalVariable(symbol, cfg)
 }
 
+// Reslice will attempt to 'reslice' a map, array or slice so that the values
+// from [start:start+count] are loaded.
+func (d *Debugger) Reslice(v *proc.Variable, start, count int, cfg proc.LoadConfig) (*proc.Variable, error) {
```

Comment:
I prefer this approach simply due to the fact that the caller doesn't have to worry about locking target, that can remain an internal implementation detail of how `Debugger` interacts with `Target`. I know it adds another layer of indirection which can be a bit annoying, but it provides convenience for the caller.

Reply:
I'm worried about the proliferation of debugger methods that just acquire a lock and call a function in proc and only exist to implement a feature of DAP.

Reply:
@aarzilli I share that concern as well, and I'm open to suggestions for refactoring to remove some layers of abstraction that we may not need. But I think my original concern still stands that having the caller control the lock seems a bit less than ideal.

Another idea could be to have a helper function within the DAP code which calls `Debugger.LockTarger/UnlockTarget` within a single function call to emulate what's happening now but not make it part of the public Debugger API. Thoughts on that?

Reply:
> But I think my original concern still stands that having the caller control the lock seems a bit less than ideal.

I don't care much about that since there's already a call to LockTarget/UnlockTarget in dap/server.go to convert goroutines efficiently.

> Another idea could be to have a helper function within the DAP code which calls Debugger.LockTarger/UnlockTarget within a single function call to emulate what's happening now but not make it part of the public Debugger API. Thoughts on that?

I don't have a preference between that and a helper in debugger.

Reply:
> > But I think my original concern still stands that having the caller control the lock seems a bit less than ideal.
> 
> I don't care much about that since there's already a call to LockTarget/UnlockTarget in dap/server.go to convert goroutines efficiently.

True, and yeah I know that exists. I just want to make it easy for the right thing to happen. That's a benefit of having an abstraction layer like `Debugger` provides, is the caller of those methods never has to think about locking/unlocking. That decision is made only once during implementation. More code == more bugs, so centralizing the mutex locking/unlocking seems safer.

> > Another idea could be to have a helper function within the DAP code which calls Debugger.LockTarger/UnlockTarget within a single function call to emulate what's happening now but not make it part of the public Debugger API. Thoughts on that?
> 
> I don't have a preference between that and a helper in debugger.

Alright, well I'm fine with this as it stands. Unless you have major concerns we can merge as is (after @polinasok +1).